Sources/eppo/EppoClient.swift

```diff
@@ -60,37 +60,48 @@ public class EppoClient {
         self.configurationStore = ConfigurationStore(requester: configurationRequester)
     }
     
-    public static func configure(
+    public static func initialize(
         apiKey: String,
         host: String = "https://fscdn.eppo.cloud",
         assignmentLogger: AssignmentLogger? = nil,
         assignmentCache: AssignmentCache? = InMemoryAssignmentCache()
-    ) -> EppoClient {
-        if instance == nil {
-            instance = EppoClient(apiKey: apiKey, host: host, assignmentLogger: assignmentLogger, assignmentCache: assignmentCache)
+    ) async throws -> EppoClient {
+        let tentativeNewInstance = EppoClient(
+            apiKey: apiKey,
+            host: host,
+            assignmentLogger: assignmentLogger, 
+            assignmentCache: assignmentCache
+        )
+
+        if let instance = sharedInstance, instance != tentativeNewInstance {
+            // If the shared instance is not the same as the new client, update it.
+            // Conditions for equality are defined in the `==` operator.
+            sharedInstance = tentativeNewInstance
+            try await sharedInstance!.loadIfNeeded()
+        } else if sharedInstance == nil {
+            // If the shared instance is nil, set it to the new client.
+            sharedInstance = tentativeNewInstance
+            try await sharedInstance!.loadIfNeeded()
         }
-        return instance!
+    
+        return try shared()
     }
     
-    public static func getInstance() throws -> EppoClient {
-        guard let instance = instance else {
+    public static func shared() throws -> EppoClient {
+        guard let instance = sharedInstance else {
             throw Errors.notConfigured
         }
         return instance
     }
     
-    public static func resetInstance() {
-        instance = nil
+    public static func resetSharedInstance() {
+        sharedInstance = nil
```

Comment:
is this only for tests? if do I'd recommend leaving a comment to explicitly point that out since this could have dangerous consequences in prod

Reply:
No - since this is a singleton we are allowing people to call this method repeatedly. Also since this is an async method people might choose to call it from app start, but there would be a race condition with the observer. Therefore they might want to put initialization next to assignment and this init will be called a lot. The readme has both of these examples.